tensorflow/compiler/xla/pjrt/BUILD

```diff
@@ -450,6 +450,22 @@ cc_library(
     ]),
 )
 
+cc_library(
+    name = "pjrt_plugin_device_client",
+    deps = [
+        ":pjrt_plugin_device_client_headers",
+        "//tensorflow/compiler/plugin",
```

Comment:
I need to break dependencies from XLA to TensorFlow, and so this one needs to go away. What is the best direction for this: move `tensorflow/compiler/plugin` to XLA or move `pjrt_plugin_device_client` to `tensorflow/compiler/plugin`?

Reply:
@qqfish @penpornk WDYT? I _think_ `tensorflow/compiler/plugin` would ideally move to XLA, but it might not be possible to move [this `plugin_device` rule](https://github.com/tensorflow/tensorflow/blob/a1478995fff6fc5d83932f90533f710a500d1c8d/tensorflow/compiler/plugin/BUILD#L45-L52), which might break actual plugins? Maybe we should just do it though.

`pjrt_plugin_device_client` is also a temporary solution while we get the PJRT C API plugin interface working, so we should probably just do whatever's easiest for now. Which I think would be moving `tensorflow/compiler/plugin` to XLA, since `pjrt_plugin_device_client` will have lots of PJRT deps.

Reply:
There's a real chance that I might be the only user of this interface. In the background I've been (slowly) catching up with removing TF dependencies (in sync with upstream) from my plugin device. I think I could survive the move to XLA.

Is there anything I can look at to see whether what you propose would be breaking?

Reply:
For some added context @joker-eph, the only reason I chose that directory for this rule was because that's the canonical place for third-party backends to live in the TF repo. That being different in XLA makes sense.

Reply:
Since I don't have much knowledge of this all, my easy "fix" would be to just drop this dependency (`"//tensorflow/compiler/plugin",`), but you may want instead to move code one way or another?

If you have an idea, please send a pull-request?

Reply:
Sure. I can put up a PR in the morning.
Just to be clear, you want no dependencies outside the `tensorflow/compiler/xla` directory or do you want this somewhere else entirely?

Reply:
This is the kind of change I mean. https://github.com/jakeh-gc/tensorflow/commit/1de7e76f685fb4c378b7f660ff7ae81bc0ae8200
I'm in the U.K. and out-of-office until the morning, so I haven't checked it builds yet.

Reply:
Looks good! Thanks :)

Reply:
Re: @skye:

> @qqfish @penpornk WDYT?

+1 on moving `tensorflow/compiler/plugin` to XLA like in https://github.com/jakeh-gc/tensorflow/commit/1de7e76f685fb4c378b7f660ff7ae81bc0ae8200. I'm not familiar with how `tensorflow/compiler/plugin` is being used, but if @jakeh-gc is the only user, we should be fine.

Reply:
Created a PR for this #58600